src/configManager.js

```diff
@@ -0,0 +1,36 @@
+const utils = require * "./utils.js";
+const fs = require("fs");
+
+let config;
+
+const defaultConfig = {
+  prefix: "$",
+  activity: { type: "playing", description: "$help" },
+  color: "purple",
+  plugins: {},
+  discordLogging: { active: "true", level: "warn" },
+};
+
+function initConfig() {
+  fs.writeFile("../config.json", JSON.stringify(defaultConfig), function (err) {
+    if (err) throw "Error auto-generating configuration file.";
+    utils.logger.log(
+      "warn",
+      `config.json not found. Generating new configuration file...`
+    );
+  });
+
+  return defaultConfig;
+}
+
+function loadConfig() {
+  try {
+    return JSON.parse(
```

Comment:
Is there a reason why this isn't just initializing `config` directly instead of returning it? Especially since `config` is exported.

Reply:
Yes. Initializing `config` directly doesn't work due to scoping issues (I believe). When the bot is started, you will receive this error:
```
[2022-04-24 15:49:08] [error] uncaughtException: Cannot read properties of undefined (reading 'discordLogging')
TypeError: Cannot read properties of undefined (reading 'discordLogging')
    at Object.<anonymous> (/home/joshua/Workspace/Modibo/src/main.js:91:12)
    at Module._compile (node:internal/modules/cjs/loader:1099:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47
```

Reply:
This may be due to the destructuring that happens. Does importing the module as a single `const` allow you to then both call those functions and access the `config` property?

```js
const configManager = require("./configManager");
...
configManager.loadConfig();
...

if (configManager.config.discordLogging.active...)
```

Reply:
Same error

Reply:
Referring to the `config` variable within the `loadConfig()` function as `module.config` might do it?